pkg/controllers/managementuser/helm/controller.go

```diff
@@ -336,6 +338,33 @@ func (l *Lifecycle) Run(obj *v3.App, template string, tempDirs *hCommon.HelmPath
 	return l.createAppRevision(obj, template, notes, false)
 }
 
+// pruneOldRevisions checks if the total number of app revisions does not exceed the set maximum.
+// If it does, the method deletes the oldest revision(s) to maintain the maximum number of revisions and no more.
+func (l *Lifecycle) pruneOldRevisions(obj *v3.App) {
+	if obj.Spec.MaxRevisionCount < 1 {
```

Comment:
If number is negative, should we skip the prune logic? This provides a way to skip garbage collection.

Reply:
We're just enforcing the default value for `MaxRevisionCount` (which can only be set once on app creation). If the number is negative, and we skip pruning, then new revisions will continue to be created without bounds, right?

Reply:
Yeah if number is negative then we don't purge revisions.

Reply:
We do now, since we set the max to 10 and check how many exist currently. So we can keep the current behavior, right?

Reply:
talked offline, it should be fine